kornia/augmentation/random_generator/_2d/color_jitter.py

```diff
@@ -71,20 +78,21 @@ def make_samplers(self, device: torch.device, dtype: torch.dtype) -> None:
         self.contrast_sampler = UniformDistribution(contrast[0], contrast[1], validate_args=False)
         self.hue_sampler = UniformDistribution(hue[0], hue[1], validate_args=False)
         self.saturation_sampler = UniformDistribution(saturation[0], saturation[1], validate_args=False)
-        self.randperm = partial(torch.randperm, device=device, dtype=dtype)
 
     def forward(self, batch_shape: Tuple[int, ...], same_on_batch: bool = False) -> Dict[str, Tensor]:
         batch_size = batch_shape[0]
-        _common_param_check(batch_size, same_on_batch)
-        _device, _dtype = _extract_device_dtype([self.brightness, self.contrast, self.hue, self.saturation])
         brightness_factor = _adapted_rsampling((batch_size,), self.brightness_sampler, same_on_batch)
         contrast_factor = _adapted_rsampling((batch_size,), self.contrast_sampler, same_on_batch)
         hue_factor = _adapted_rsampling((batch_size,), self.hue_sampler, same_on_batch)
         saturation_factor = _adapted_rsampling((batch_size,), self.saturation_sampler, same_on_batch)
+
+        # perform random permutation of the order of the color adjustments
+        random.shuffle(self.randperm_list)
+
         return {
-            "brightness_factor": brightness_factor.to(device=_device, dtype=_dtype),
```

Comment:
just to confirm, we didn't need this cast here, in favor of using base generator move/cast `to` ?

Reply:
right